### DISCUSSION TITLE
Issue when adding a custom chain to the metamask wallet from mobile device

### DISCUSSION THREAD

**Discussion by kacpergumieniuk** [rainbow-me/rainbowkit#1928] [Q&A]
Hey, I'm building an app basing on custom chain and I've implemented rainbowkit recently. Now everything works fine, but when it comes to mobie device and user flow where he does not have custom chain defined in his device, changing network from the browser does not work, it simply opens metamask but calling no action. It works from built-in metamask browser, but we all know that user won't use that. If you need any details, i will provide:)

**Comment by yashsisodiya** [comment]
Facing the Same issue did you got any solutions

**Reply by kacpergumieniuk** [reply]
nope

**Comment by 0xmkh** [comment]
@kacpergumieniuk With MetaMask on mobile we've had tons of comments on the issue you're seeing. This is because we use WalletConnect to initiate a session in MetaMask on mobile. The best thing to do for now is to refresh the page, closing and re-opening MetaMask app or even use MetaMask's browser wallet instead.

The WalletConnect and MetaMask team is working to fix this issue 🙏
